src/go/cmd/http-relay-server/server/server.go

```diff
@@ -200,6 +200,7 @@ func newBackendContext(r *http.Request) (*backendContext, error) {
 
 func (s *Server) health(w http.ResponseWriter, r *http.Request) {
 	if err := s.b.Healthy(); err != nil {
+		slog.Error("Health check failed", ilog.Err(err))
```

Comment:
Since the health check just locks a mutex, I'd be worried that this will never execute: Either the server is deadlocked and Healthy() never returns, or it's healthy and it returns nil.

I think we'd need to detect the deadlock by observing the case where either Lock() or Healthy() hangs for longer than 10 seconds and print the log statement in that case. Not sure if there's a race with Kubernetes killing the pod so maybe 8-9s is safer.

Reply:
https://github.com/googlecloudrobotics/core/pull/359/commits/5ba9554156264e79672f39d09c81185ed5697da8 Added a 8 secs timer to print a warning.

Reply:
Really clean solution, I was imagining something with channels and selects and so on. Thank you!!!